tests/conftest.py

```diff
@@ -62,6 +62,43 @@ def _generate_app(
 ]
 
 
+@pytest.fixture(autouse=True, scope="function")
+def forbid_external_commands(monkeypatch):
```

Comment:
NOTE: these fixtures actually don't work for the CLI tests in this PR since they are run via Runner which launches `aiidalab` as external process. But they will be useful later for unit-style tests.

Reply:
I am not quite understand how this would work. If the unit test is running with this fixture loaded, will it monitor the system call of `pip install` etc?

Reply:
In this fixture I am using `monkeypatch` to replace the python functions in `aiidalab.utils` module. If the unit tests ends up calling them, this fixture will raise an exception. 

I'll see if this makes sense once I start writing the tests, I will change it if not.